src/mmds/src/lib.rs

```diff
@@ -254,6 +259,33 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_unsupported_http_method() {
+        // Test invalid HTTP POST method. This is not supported by `micro-http`.
+        let request = b"POST http://169.254.169.254/ HTTP/1.0\r\n\r\n";
+        let mut expected_response = Response::new(Version::Http10, StatusCode::NotImplemented);
+        expected_response.set_body(Body::new("Unsupported HTTP method.".to_string()));
+        let actual_response = parse_request(request);
+        assert!(expected_response.status() == actual_response.status());
+        assert!(expected_response.body().unwrap() == actual_response.body().unwrap());
+
+        // Test `PUT`. Supported by `micro-http`, but not supported by MMDS.
+        let request = b"PUT http://169.254.169.254/ HTTP/1.0\r\n\r\n";
+        let mut expected_response = Response::new(Version::Http10, StatusCode::NotImplemented);
+        expected_response.set_body(Body::new("Unsupported HTTP method.".to_string()));
+        let actual_response = parse_request(request);
+        assert!(expected_response.status() == actual_response.status());
+        assert!(expected_response.body().unwrap() == actual_response.body().unwrap());
+
+        // Test `PATCH`. Supported by `micro-http`, but not supported by MMDS.
+        let request = b"PATCH http://169.254.169.254/ HTTP/1.0\r\n\r\n";
+        let mut expected_response = Response::new(Version::Http10, StatusCode::NotImplemented);
+        expected_response.set_body(Body::new("Unsupported HTTP method.".to_string()));
+        let actual_response = parse_request(request);
+        assert!(expected_response.status() == actual_response.status());
+        assert!(expected_response.body().unwrap() == actual_response.body().unwrap());
+    }
```

Comment:
Please also add a test with DELETE and HEAD, and if possible send an invalid HTTP method like "let request = b"NOMETHOD http:////169.254.169.254/ HTTP/1.0\r\n\r\n"

Reply:
I've added tests for all HTTP methods and a presumably invalid HTTP method.